hs_composite_resource/models.py

```diff
@@ -195,35 +626,173 @@ def supports_delete_folder_on_zip(self, original_folder):
         if not full_path.startswith(self.file_path):
             full_path = os.path.join(self.file_path, full_path)
 
-        if self.is_federated:
-            res_file_objects = self.files.filter(
-                object_id=self.id,
-                fed_resource_file__startswith=full_path).all()
-        else:
-            res_file_objects = self.files.filter(
-                object_id=self.id,
-                resource_file__startswith=full_path).all()
+        # get all resource files at full_path and its sub-folders
+        res_file_objects = ResourceFile.list_folder(self, full_path)
 
         # check any logical file associated with the resource file supports deleting the folder
         # after its zipped
         for res_file in res_file_objects:
-            if not res_file.logical_file.supports_delete_folder_on_zip:
-                return False
-
+            if res_file.has_logical_file:
+                if not res_file.logical_file.supports_delete_folder_on_zip:
+                    return False
         return True
 
     def get_missing_file_type_metadata_info(self):
         # this is used in page pre-processor to build the context
-        # so that the landing page can show what metadata items are missing for each logical file
+        # so that the landing page can show what metadata items are missing for each
+        # logical file/aggregation
         metadata_missing_info = []
         for lfo in self.logical_files:
             if not lfo.metadata.has_all_required_elements():
-                file_path = lfo.files.first().short_path
                 missing_elements = lfo.metadata.get_required_missing_elements()
-                metadata_missing_info.append({'file_path': file_path,
+                metadata_missing_info.append({'file_path': lfo.aggregation_name,
                                               'missing_elements': missing_elements})
         return metadata_missing_info
 
+    def update_coverage(self):
+        """Update resource spatial and temporal coverage based on the corresponding coverages
+        from all the contained aggregations (logical file) only if the resource coverage is not
+        already set"""
+
+        # update resource spatial coverage only if it is empty
+        if self.metadata.spatial_coverage is None:
```

Comment:
None does not include empty.  Perhaps this line should be `if self.metadata.spatial_coverage:`

Reply:
spatial_coverage can be either None or not None. Will update the code comment to reflect that.